lib/internal/util/comparisons.js

```diff
@@ -107,8 +108,9 @@ function strictDeepEqual(val1, val2, memos) {
     if (val1.length !== val2.length) {
       return false;
     }
-    const keys1 = getOwnNonIndexProperties(val1);
-    if (keys1.length !== getOwnNonIndexProperties(val2).length) {
+    const keys1 = getOwnNonIndexProperties(val1, ONLY_ENUMERABLE);
+    const keys2 = getOwnNonIndexProperties(val2, ONLY_ENUMERABLE);
+    if (keys1.length !== keys2.length) {
```

Comment:
Unrelated optimization idea: would it make sense to return only the length from V8 rather than the properties themselves in the case of keys2?